src/quantuminspire/qiskit/qi_job.py

```diff
@@ -66,12 +77,14 @@ def submit(self) -> None:
             raise JobError('Job has already been submitted!')
         self._job_id = self._backend.run(self._qobj)
 
-    def result(self, timeout: Optional[float] = None, wait: float = 0.5) -> QIResult:
+    def result(self, timeout: Optional[float] = None, wait: float = 0.5, only_latest_run: bool = True) -> QIResult:
```

Comment:
Similar to the comment on `get_experiment_results`, you might want to consider removing the additional argument, and in stead create a new function `all_results_from_project`, or something similarly named.

Reply:
I can do that but this QIJob class is subclassed from a Qiskit class BaseJob where the interface is specified for getting results (job.result()) in Qiskit. I want to conform to this interface instead of extending it. Of course I can extend the interface and add a new all_results_from_project but the users won't easily find it and use it because they are known to the Qiskit interface for getting back the results. It's described this way in the Qiskit documentation etc. Only the ones who dive into it a bit and really understand how Quantum Inspire works with projects and jobs, may change (one of) the parameter of result().

Reply:
Hmm. I'm not sure that this argument holds. You are already extending the interface by introducing a new boolean flag as function parameter. A flag parameter to make the function behave differently is actually a well-known code smell. The recommended refactoring for that is to create a new function for the different behavior.
And in both cases (flag parameter or new function) the user must have done some research before they can use it.

Reply:
I'll add another method for getting the results for all runs: results_all_runs